lib/Service/Classification/ImportanceClassifier.php

```diff
@@ -279,21 +417,25 @@ private function getFeaturesAndImportance(Account $account,
 	/**
 	 * @param Account $account
 	 * @param Message[] $messages
+	 * @param LoggerInterface $logger
 	 *
 	 * @return bool[]
+	 *
 	 * @throws ServiceException
 	 */
-	public function classifyImportance(Account $account, array $messages): array {
-		$estimator = null;
+	public function classifyImportance(Account $account,
+		array $messages,
+		LoggerInterface $logger): array {
+		$pipeline = null;
 		try {
-			$estimator = $this->persistenceService->loadLatest($account);
+			$pipeline = $this->persistenceService->loadLatest($account);
 		} catch (ServiceException $e) {
-			$this->logger->warning('Failed to load importance classifier: ' . $e->getMessage(), [
+			$logger->warning('Failed to load persisted estimator and extractor: ' . $e->getMessage(), [
 				'exception' => $e,
 			]);
 		}
 
-		if ($estimator === null) {
+		if ($pipeline === null) {
```

Comment:
Previously this condition was only met for new accounts where no training has happened yet. Now servers with no memory cache will always run into the rule-based classifier. I don't think this is desirable, because results will be bad. The rules based classifier is only meant to cold start the classification.
How about we skip classification all together if there is no distributed cache available?

Reply:
Or, instead of skipping, we generate a pipeline on the fly? It's slow but gives good results

Reply:
I implemented training on-the-fly as it is rather quick. If this also fails (because the user has not enough messages), the classification will be skipped. Otherwise, the rule based classifier will be used to cold start the pipeline.

Reply:
Perfect, thank you.